src/unitxt/operators.py

```diff
@@ -658,6 +665,65 @@ def process(self, stream: Stream, stream_name: str = None) -> Generator:
                 yield instance
 
 
+class FilterByListsOfValues(SingleStreamOperator):
+    """
+    Filters a stream, yielding only instances that  whose field values are included in the specified value lists.
+
+    Args:
+        values (Dict[str, Any]): For each field, the list of values that instances should match to be included in the output.
+    """
+
+    values: Dict[str, Any]
+
+    def verify(self):
+        for key, value in self.values.items():
+            if not isinstance(value, list):
+                raise ValueError(f"The filter for key ('{key}') in FilterByListsOfValues is not a list but '{value}'")
+
+    def process(self, stream: Stream, stream_name: str = None) -> Generator:
+        for instance in stream:
+            if any(instance[key] in value for key, value in self.values.items()):
+                yield instance
+
+
+class Intersect(FieldOperator):
+    """
+    Intersects the value of a field, which must be a list, with a given list
+    Args:
+        allowed_values (list) - list to intersect
+    """
+
+    allowed_values: List[Any]
+
+    def verify(self):
+        if not isinstance(self.allowed_values, list):
+            raise ValueError(f"The allowed_values is not a list but '{self.allowed_values}'")
+
+    def process_value(self, value: Any) -> Any:
+        if not isinstance(value, list):
+            raise ValueError(f"The value in field is not a list but '{value}'")
```

Comment:
what if `process_every_value` is `True`? In this case the `value` is only one of the values in the field, and not the entire value of the field. In this case, shouldn't we allow list values? (e.g. the field value is [ ["a", "b"], "c" ].

I think at least the error message should be f"The processed value is not a list but '{value}'".

Reply:
I decided not to allow nested list (process_every_value is True) .   It's too complicated.